### PR TITLE
[WIP] Update secp256k1 to btc v0.6.0

### PR DESCRIPTION
**Pull request by joii2020**
Upgrade secp256k1 to use the [official](https://github.com/bitcoin-core/secp256k1/tree/v0.6.0) version.
Size: 121K -> 143K
Cycles 1.43M -> 1.40M (Take the average of multiple `test_btc_success`)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,26 +25,18 @@ all: build/omni_lock build/always_success
...
-	gcc -I deps/secp256k1-20210801/src -I deps/secp256k1-20210801 -o $@ $<
-
-
-$(SECP256K1_SRC_20210801):
-	cd deps/secp256k1-20210801 && \
-		./autogen.sh && \
-		CC=$(CC) LD=$(LD) ./configure --with-bignum=no --enable-ecmult-static-precomputation --enable-endomorphism --enable-module-recovery --host=$(TARGET) && \
-		make src/ecmult_static_pre_context.h src/ecmult_static_context.h
-
+	gcc -I deps/secp256k1-20210801/src -o $@ $<
```
Comment (XuJiandong) [inline Makefile:39]: Why are these line removed? New secp256k1 lib don't require config?

Comment (joii2020) [reply Makefile:39]: [precomputed_ecmult.c](https://github.com/bitcoin-core/secp256k1/blob/0cdc758a56360bf58a851fe91085a327ec97685a/src/precomputed_ecmult.c) already exists in the repositorie and does not need to be regenerated.

Comment (XuJiandong) [reply Makefile:39]: it is too large, about 2.3M bytes. We need a smaller one.

Comment (joii2020) [reply Makefile:39]: But the exported secp256k1_data_20210801 is still 1M (1048576).
The size still needs to be controlled at around 20k? (ckb-auth)

Comment (XuJiandong) [reply Makefile:39]: it's ok to have same size as before (1M).
